src/rdkafka_request.c

```diff
@@ -2431,6 +2434,15 @@ rd_kafka_error_t *rd_kafka_ListGroupsRequest(rd_kafka_broker_t *rkb,
                 rd_kafka_buf_finalize_arraycnt(rkbuf, of_GroupsArrayCnt, i);
         }
 
+        if (ApiVersion >= 5) {
+                size_t of_GroupsArrayCnt =
+                    rd_kafka_buf_write_arraycnt_pos(rkbuf);
```

Comment:
This we do only if we don't know the exact value of count. Right now we know the value so we can just put that value here instead of filling later.

Reply:
agreed.